django/db/models/expressions.py

```diff
@@ -843,13 +843,19 @@ def as_sql(self, compiler, connection, template=None, case_joiner=None, **extra_
         template_params.update(extra_context)
         case_parts = []
         sql_params = []
+        from django.db.models.sql.datastructures import EmptyResultSet
```

Comment:
Maybe, I don't feel strongly about the need to move it just to avoid some circular imports. On the other hand, now the `Case` needs it, is there a chance that other expressions might need it too?